utilities/merge_operators/collection/multi_operand_list.h

```diff
@@ -0,0 +1,447 @@
+  //     }
+  //   }
+
+  //   if (!in_multi_
+  //       && it_operand_list_ != it_operand_list_end_) {
+  //     // read next from underlying operand_list
+
+  //     if (++it_operand_list_ == it_operand_list_end_) {
+  //       operand = nullptr;
+  //     } else {
+  //       operand = &(*it_operand_list_);
+
+  //       // is multi Collection Operation?
+  //       if ((*operand)[0] == CollectionOperation::_kMulti) {
+  //         in_multi_ = true;
+  //         multi_operand_ = operand;
+  //         build_multi_operand_index();
+  //         multi_operand_index_offset_ = 0;
+
+  //         operand = get_multi_operand_operand(multi_operand_index_offset_);
+  //       }
+  //     }
+  //   }
+
+  //   return operand;
+  // }
+  // TODO(AR) temp - see https://github.com/facebook/rocksdb/issues/3655
+  // handles the case where previous PartialMergeMulti returns a no-op
+  const Slice* next_operand() {
+    const Slice* operand = nullptr;
+
+    if (in_multi_) {
+      if (has_next_multi_operand()) {
+
+        // cleanup memory from previous operand of multi
+        delete current_operand_;
+
+        // return next from multi
+        operand = get_multi_operand_operand(++multi_operand_index_offset_);
+
+      } else {
+        // end of previous multi
+        in_multi_ = false;
+        multi_operand_ = nullptr;
+        delete current_operand_;  // free the memory allocated for the last multi
+      }
+    }
+    
+    if (!in_multi_
+        && it_operand_list_ != it_operand_list_end_) {
+      // read next from underlying operand_list
+      
+      while (++it_operand_list_ != it_operand_list_end_) {
+        operand = &(*it_operand_list_);
+
+        // handles the case where previous PartialMergeMulti returns a no-op
+        if (operand->size() == 0) {
+            continue; // this is empty, attempt next
+        }
+
+        // is multi Collection Operation?
+        if ((*operand)[0] == CollectionOperation::_kMulti) {
+          in_multi_ = true;
+          multi_operand_ = operand;
+          build_multi_operand_index();
+          multi_operand_index_offset_ = 0;
+          
+          operand = get_multi_operand_operand(multi_operand_index_offset_);
+        }
+
+        break;
+      }
+
+      if (it_operand_list_ == it_operand_list_end_) {
+        operand = nullptr;
+      }
+    }
+
+    return operand;
+  }
+
+  bool has_next_multi_operand() const {
+    return in_multi_
+        && multi_operand_index_offset_ + 1 < multi_operand_index_.size() ;
+  }
+
+  const Slice* get_multi_operand_operand(const size_t multi_operand_index_offset) const {
+    const char* multi_operand_ptr = multi_operand_index_.at(multi_operand_index_offset);
+    const uint32_t records_len = DecodeFixed32(multi_operand_ptr);
+    multi_operand_ptr += sizeof(uint32_t);
+
+    const Slice* operand = new Slice(multi_operand_ptr, sizeof(CollectionOperation) + records_len);
+    return operand;
+  }
+
+  // const Slice* prev_operand() {
```

Comment:
Maybe remove all the commented code.

Reply:
That's there as the correct code to be reinstated once https://github.com/facebook/rocksdb/issues/3655 is solved